packages/restapi/src/lib/spaceV2/acceptInvite.ts

```diff
@@ -0,0 +1,261 @@
+                    ) {
+                        // destroy connection to only the current peer
+                        peerConnection?.destroy();
+                        peerConnection = null;
+                        this.setSpaceV2Data((oldData) => {
+                            return produce(oldData, (draft) => {
+                                const incomingIndex = getIncomingIndexFromAddress(
+                                    oldData.incomingPeerStreams,
+                                    recipientAddress
+                                );
+                                draft.incomingPeerStreams.splice(incomingIndex, 1);
+                            });
+                        });
+                    }
+                    if (
+                        parsedData?.details?.type === SPACE_DISCONNECT_TYPE.STOP
+                    ) {
+                        // destroy connection to all the peers
+                        for (const connectedAddress in this.getConnectedPeerIds()) {
+                            // TODO: refactor
+                            // this.getPeerConnection(connectedAddress)?.destroy();
+                            this.setPeerConnection(connectedAddress, undefined);
+                        }
+                    }
+
+                    if (
+                        parsedData?.details?.type === SPACE_DISCONNECT_TYPE.STOP
+                    ) {
+                        // destroy the local stream
+                        if (this.data.local.stream) {
+                            endStream(this.data.local.stream);
+                        }
+
+                        // reset the state
+                        this.setSpaceV2Data(() => initSpaceV2Data);
+                    }
+                }
+            } else {
+                onReceiveMessage(data);
+            }
+        });
+
+        peerConnection.on(
+            'stream',
+            (currentStream: MediaStream) => {
+                console.log('received incoming stream', currentStream);
+                const pendingStreamIndex = getIncomingIndexFromAddress(
+                    this.data.pendingPeerStreams,
+                    recipientAddress
+                );
+
+                // Here, we can handle if we want to merge stream or anything
+                // this.onReceiveStream(
+                //   currentStream,
+                //   recipientAddress,
+                //   this.data.pendingPeerStreams[pendingIndex].audio
+                // );
+
+                // remove stream from pendingPeerStreams and add it to incomingPeerStreams
+                this.setSpaceV2Data((oldData) => {
+                    return produce(oldData, (draft) => {
+                        draft.incomingPeerStreams.push(draft.pendingPeerStreams[pendingStreamIndex]);
+                        draft.pendingPeerStreams.splice(pendingStreamIndex, 1);
+
+                        const incomingStreamIndex = getIncomingIndexFromAddress(
```

Comment:
for optimized code, get the peer from pendingPeerStream in a variable and update the stream in it
and then directly push it into incomingPeerStreams

in this way, incomingPeerStreams won't be traversed again to find the index